### ISSUE TITLE
[Bug] `InterpolatedLazyTensor` cannot handle rectangular `base_lazy_tensors`

### ISSUE THREAD

**Issue by wecacuee** [cornellius-gp/gpytorch#900]
# 🐛 Bug

`InterpolatedLazyTensor` results in error when base_lazy_tensor is not square

## To reproduce

** Code snippet to reproduce **
```python
    itplzt = InterpolatedLazyTensor(NonLazyTensor(torch.rand(2, 3)))
    res = itplzt @ torch.eye(3)
    assert res.shape == (2, 3)
```

** Stack trace/error message **
```
(py37) vdhiman@dwarf:~/wrk/BayesCBF_ws/BayesCBF$ python tests/test_interpolated_lazy_tensor.py 
Traceback (most recent call last):
  File "tests/test_interpolated_lazy_tensor.py", line 30, in <module>
    test_interpolated_lazy_tensor()
  File "tests/test_interpolated_lazy_tensor.py", line 7, in test_interpolated_lazy_tensor
    res = itplzt @ torch.eye(3)
  File "/home/vdhiman/wrk/gpytorch/gpytorch/lazy/lazy_tensor.py", line 1731, in __matmul__
    return self.matmul(other)
  File "/home/vdhiman/wrk/gpytorch/gpytorch/lazy/interpolated_lazy_tensor.py", line 393, in matmul
    right_interp_res = left_t_interp(self.right_interp_indices, self.right_interp_values, tensor, base_size)
  File "/home/vdhiman/wrk/gpytorch/gpytorch/utils/interpolation.py", line 202, in left_t_interp
    values = rhs.unsqueeze(-2) * interp_values.unsqueeze(-1)
RuntimeError: The size of tensor a (3) must match the size of tensor b (2) at non-singleton dimension 0
```

## Expected Behavior

Expect res to be a Tensor of size (2,3)

## System information

- GPyTorch Version 0.3.5 
- PyTorch Version 1.2.0
- Ubuntu 16.04.6 LTS

## Additional information
The test cases in `test_interpolated_lazy_tensor.py` carefully construct PSD, square symmetric `base_lazy_tensor`s but the constructor or the documentation does not say so. I am specifically getting errors when `InterpolatedLazyTensor` is being constructed at `lazy_tensor.py:307`. In that case, the `baze_lazy_tensor` is not square.
Adding the following test case to `test_interpolated_lazy_tensor.py` fails almost all test cases
```python
class TestInterpolatedLazyTensorRectangular(LazyTensorTestCase, unittest.TestCase):
    def create_lazy_tensor(self):
        itplzt = InterpolatedLazyTensor(NonLazyTensor(torch.rand(2, 3)))
        return itplzt

    def evaluate_lazy_tensor(self, lazy_tensor):
        return lazy_tensor.base_lazy_tensor.tensor
```

**Comment by jacobrgardner** [comment]
I think this may just be a bug in how we handle the case where you don't specify the interpolation components at all, which you aren't in your code (num_rows is used for both):
https://github.com/cornellius-gp/gpytorch/blob/508155543637d88d046cb93979ad6745e9307159/gpytorch/lazy/interpolated_lazy_tensor.py#L60

I believe if they are specified, it may work for rectangular matrices? I'll look in to seeing if this is easy to fix -- I honestly don't remember if we intended ILT to work for rectangular matraices.

**Comment by wecacuee** [comment]
Thanks for responding.

I think that is a bug. But that is not the bug that I am facing for my usage. Maybe, I will open another ticket for it.

**Comment by jacobrgardner** [comment]
Fixed by #906 

`_matmul` only gets run in nograd contexts / pytorch Functions. `matmul` actually calls the `Matmul` pytorch function.
